wsi/drm/drmdisplay.cpp

```diff
@@ -205,12 +205,15 @@ bool DrmDisplay::ConnectDisplay(const drmModeModeInfo &mode_info,
   GetDrmObjectProperty("CRTC_ID", connector_props, &crtc_prop_);
   GetDrmObjectProperty("Broadcast RGB", connector_props, &broadcastrgb_id_);
   GetDrmObjectProperty("DPMS", connector_props, &dpms_prop_);
+  GetDrmObjectProperty("max bpc", connector_props, &max_bpc_prop_);
```

Comment:
[MINOR] For consistency, suggest "max_bpc" to be in all caps.

Reply:
We can't change this , the convention is as per Kernel implementation

Reply:
> @harishkrupo I do have some comments regarding this patch:
> 
> 1. The purpose of the patch is to crop the max_bpc which in essence will reduce the color gamut but it is a workaround for faulty dongles.  Upon setting the max_bpc, I don't see how these two variables are getting updated:
>    ```
>       old_connector_state->max_requested_bpc
>       new_connector_state->max_requested_bpc
>    ```

@MicheleL-Intel The patch that I shared is a bit old. The latest atm is v13:
https://patchwork.freedesktop.org/patch/256122/

IIRC, the new state would be set here:
```
diff --git a/drivers/gpu/drm/drm_atomic_uapi.c b/drivers/gpu/drm/drm_atomic_uapi.c
index a22d6f269b07..7492819c81ce 100644
--- a/drivers/gpu/drm/drm_atomic_uapi.c
+++ b/drivers/gpu/drm/drm_atomic_uapi.c
@@ -761,6 +761,8 @@  static int drm_atomic_connector_set_property(struct drm_connector *connector,
 
 		return set_out_fence_for_connector(state->state, connector,
 						   fence_ptr);
+	} else if (property == connector->max_bpc_property) {
+		state->max_requested_bpc = val;
 	} else if (connector->funcs->atomic_set_property) {
 		return connector->funcs->atomic_set_property(connector,
 				state, property, val);
```
Then it would be compared with the old state (i.e. the current state before we finish the atomic commit)

> 2. Just the basics ... there are unchecked pointers not in the patch but elsewhere.  For example, local variables created in drm_atomic_check_only() are not validated before they are passed into drm_atomic_plane_check() and drm_atomic_connector_check().

I am not sure about this. Probably the same reason we don't check gpu_fd_ everywhere we use it. You could reach out to the ML if you feel that the checks are required :)